common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -55,6 +77,43 @@ trait Logging {
     log_
   }
 
+  implicit class LogStringContext(val sc: StringContext) {
+    def log(args: Any*): (String, Option[Instance]) = {
+      val processedParts = sc.parts.iterator
+      val sb = new StringBuilder(processedParts.next())
+      lazy val map = new java.util.HashMap[String, String]()
+
+      args.foreach { arg =>
+        arg match {
```

Comment:
you can skip this line and the `arg =>` on the end of the previous line and just list the cases directly instead

Reply:
There is another code block at the end of the loop:
```
        if (processedParts.hasNext) {
          sb.append(processedParts.next())
        }
```
I am trying to build both the map and the string in one loop.